### ISSUE TITLE
Database errors after activating

### ISSUE THREAD

**Issue by FlorianBrinkmann** [inpsyde/multilingual-press#246]
Hi!

I activated the current Version 2.4.8 on a 4.7 WordPress multisite install and get the following 3 database errors:

```
DESCRIBE wp_mlp_languages;
Table 'db.wp_mlp_languages' doesn't exist

DESCRIBE wp_multilingual_linked;
Table 'db.wp_multilingual_linked' doesn't exist

DESCRIBE wp_mlp_site_relations;
Table 'db.wp_mlp_site_relations' doesn't exist
```

After reloading the page with the installed plugins I get the following errors:

```
ALTER TABLE wp_mlp_languages ADD KEY `` (`http_name`)
Incorrect index name ''

ALTER TABLE wp_multilingual_linked ADD KEY `` (`ml_blogid`,`ml_elementid`)
Incorrect index name ''

ALTER TABLE wp_mlp_site_relations ADD KEY `` (`site_1`,`site_2`)
Incorrect index name ''
```

**Comment by FlorianBrinkmann** [comment]
Now I tried the code here from GitHub and with that I only get the first set of errors (Table doesn’t exist). After reloading the page, no further errors.

**Comment by tfrommen** [comment]
Hi!

Well, there first database errors are not (really) the fault of MultilingualPress, but rather due to `dbDelta()` being used to _alter_ a table that does not yet exist, see [this post](https://www.storelocatorplus.com/turn-off-dbdelta-describe-errors/), for example, or google for "dbdelta error describe" or so. :)
It's somewhat annoying, yes, but the result is like we want it to be: the tables have been created successfully.
And once the tables are there, `dbDelta()` won't trigger any errors anymore, so this is only the case when **installing** MultilingualPress (not on any subsequent re-activations).

The second set of errors have been introduced by the changed `KEY` (or ´INDEX`) behavior of WordPress 4.6, I believe.
I definitely fixed this for `master` already (e.g., [here](https://github.com/inpsyde/multilingual-press/blob/6b091f6424bcec45ae4a6841e3dacbd57d58f90b/src/Database/Table/LanguagesTable.php#L1827)), and I thought I also did so in a patch release for the 2.4 branch - which I obviously did not (yet). Sorry.

Expect a release for this by next week. :)

Cheers,
Thorsten

**Comment by FlorianBrinkmann** [comment]
Hi Thorsten,

thanks for the explanation!

> Expect a release for this by next week. :)

That’s great, thanks for your work :)

Have a nice weekend!
Florian

**Comment by tfrommen** [comment]
Hi Florian,

can you confirm the latest changes in [the `2.4` branch](https://github.com/inpsyde/multilingual-press/tree/2.4) work for you? You can also just grab the [ZIP file](https://github.com/inpsyde/multilingual-press/archive/2.4.zip).

Just let me know and a release will follow within the next days.

Thanks,
Thorsten

**Comment by FlorianBrinkmann** [comment]
Hi Thorsten,

looks good, no `Incorrect index name` errors! :)

Cheers,
Florian

**Comment by tfrommen** [comment]
Closed with 391109f7caa361b131bc7b9df6f19e3d58d197b7.
